src/API/Reports/Products/DataStore.php

```diff
@@ -526,4 +527,90 @@ protected function initialize_queries() {
 		$this->subquery->add_sql_clause( 'from', self::get_db_table_name() );
 		$this->subquery->add_sql_clause( 'group_by', 'product_id' );
 	}
+
+	/**
+	 * Possible add filter to insert product template.
+	 */
+	public static function possibly_add_product_template() {
+		if ( isset( $_GET['post_type'] ) && 'product' === wc_clean( wp_unslash( $_GET['post_type'] ) ) && isset( $_GET['product_template'] ) ) { // phpcs:ignore CSRF ok, sanitation ok.
+			// default_excerpt is last filter called in get_default_post_to_edit.
+			add_filter( 'default_excerpt', array( __CLASS__, 'insert_product_template' ), 10, 2 );
+			// add_action('wp_insert_post', array(__CLASS__, 'insert_product_template_action'), 10, 2);
```

Comment:
This is not set in stone, but I currently added two options (scratch one of them), these both get triggered when the post-new page gets loaded with a `product_template` param. The third option is scratching this part, creating an REST endpoint for creating a product template, and then redirecting to the product edit page (in all cases the bulk of the ProductCSVTemplateImporter logic would stay the same).

The current add filter option:
- default_excerpt is the last filter called before `get_default_post_to_edit` finishes, used by `post-new.php`.
- Updates the post with the template
- Redirects to the posts edit page and exits the original process.

The second action option using `wp_insert_post`:
- Called when `get_default_post_edit` creates the initial post
- Updates the post
- The title, description, and excerpt are all set after this within `get_default_post_to_edit`, using filters
- So it adds filters for the description and excerpt (see line 566) to use the correct post data

Reply:
Curious - would it be simpler to import the appropriate sample product (as a new product) and redirect to the edit screen instead?

Reply:
Yes, if we kept the individual csv files, as (I believe, not 100% on this) the default `class-wc-product-csv-importer` imports everything within the file.
We might want the individual csv files anyway, as the sample-data has it set to auto publish.